Sources/System/MachPort.swift

```diff
@@ -0,0 +1,327 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+#if $MoveOnly && (os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
+
+import Darwin.Mach
+
+public protocol MachPortRight {}
+
+private func machPrecondition(
+    file: StaticString = #file,
+    line: UInt = #line,
+    _ body: @autoclosure () -> kern_return_t
+) {
+    let kr = body()
+    let expected = KERN_SUCCESS
+    precondition(kr == expected, file: file, line: line)
+}
+
+public enum Mach {
+    @_moveOnly
+    public struct Port<RightType: MachPortRight> {
```

Comment:
The reason why most types in System are `@frozen` is because they are defined to be layout-compatible wrappers around C types (and often are even `RawRepresentable` with the C type). `FilePath` is not frozen because it's a more complex, higher level conceptual type and it's not a big deal to pass it indirectly.

Is there any downside to passing this around indirectly? Is there any desire to be able to reason about its layout by, e.g., having an Array of these in contiguous memory without indirection?